frontend/vue/components/AmplitudeDisk/AmplitudeAdditionWidget.vue

```diff
@@ -0,0 +1,159 @@
+<template>
+  <div
+    class="amplitude-addition-widget"
+  >
+    <AmplitudeDiskWithValues
+      class="amplitude-addition-widget__disk amplitude-addition-widget__disk_a"
+      :magnitude="internalAmplitudeA.magnitude"
+      :phase="internalAmplitudeA.phase"
+      :is-interactive="true"
+      @updateAmplitude="updateInternalAmplitudeA"
+    />
+    <span class="amplitude-addition-widget__math-sign">+</span>
+    <AmplitudeDiskWithValues
+      class="amplitude-addition-widget__disk amplitude-addition-widget__disk_b"
+      :magnitude="internalAmplitudeB.magnitude"
+      :phase="internalAmplitudeB.phase"
+      :is-interactive="true"
+      @updateAmplitude="updateInternalAmplitudeB"
+    />
+    <span class="amplitude-addition-widget__math-sign">=</span>
+    <AmplitudeAdditionDiskWithValues
+      class="amplitude-addition-widget__disk amplitude-addition-widget__disk_result"
+      :class="{
+        'amplitude-addition-widget__disk_result_highlight-a': highlightedA,
+        'amplitude-addition-widget__disk_result_highlight-b': highlightedB
+      }"
+      :magnitude-a="internalAmplitudeA.magnitude"
+      :phase-a="internalAmplitudeA.phase"
+      :magnitude-b="internalAmplitudeB.magnitude"
+      :phase-b="internalAmplitudeB.phase"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue-demi'
+import AmplitudeDiskWithValues from './AmplitudeDiskWithValues.vue'
+import AmplitudeAdditionDiskWithValues from './AmplitudeAdditionDiskWithValues.vue'
+import { Amplitude } from './amplitude'
+
+export default defineComponent({
+  name: 'AmplitudeAdditionDisk',
+  components: {
+    AmplitudeAdditionDiskWithValues,
+    AmplitudeDiskWithValues
+  },
+  props: {
+    magnitudeA: {
+      type: Number,
+      required: false,
+      default: 0.8
+    },
+    phaseA: {
+      type: Number,
+      required: false,
+      default: 30
+    },
+    magnitudeB: {
+      type: Number,
+      required: false,
+      default: 0.6
+    },
+    phaseB: {
+      type: Number,
+      required: false,
+      default: 190
+    }
+  },
+  data () {
+    return {
+      internalAmplitudeA: { phase: 30, magnitude: 1 },
+      internalAmplitudeB: { phase: 30, magnitude: 1 },
+      highlightTimeoutA: -1,
+      highlightTimeoutB: -1,
+      highlightedA: false,
+      highlightedB: false
+    }
+  },
+  mounted () {
+    this.updateInternalAmplitudeA({ phase: this.phaseA, magnitude: this.magnitudeA } as Amplitude)
+    this.updateInternalAmplitudeB({ phase: this.phaseB, magnitude: this.magnitudeB } as Amplitude)
+    this.highlightedA = false
+    this.highlightedB = false
+  },
+  methods: {
+    updateInternalAmplitudeA (amplitude: Amplitude) {
+      if (this.highlightTimeoutA !== -1) {
+        clearTimeout(this.highlightTimeoutA)
+      }
+      this.highlightedA = true
+      this.highlightTimeoutA = setTimeout(this.removeHighlightA, 1000)
```

Comment:
@korgan00 there are some build failures (related to Typescript) with the various timeout variables. for example:

`TS2322: Type 'Timeout' is not assignable to type 'number'.`

Reply:
Sorry, it was working with the auto-refresh in dev.
The issue is related to node. It has an inner setTimeout and it has a different return type than the browser one.
It should be fixed now.